clients/pkg/logentry/stages/pipeline.go

```diff
@@ -74,6 +74,22 @@ func RunWith(input chan Entry, process func(e Entry) Entry) chan Entry {
 	return out
 }
 
+func RunWithSkip(input chan Entry, process func(e Entry) (Entry, bool)) chan Entry {
+	out := make(chan Entry)
+	go func() {
+		defer close(out)
+		for e := range input {
+			ee, skip := process(e)
```

Comment:
see above, but since we're running process async, do we need to put any locking around the `partialLines` slices it's modifying?